### DISCUSSION TITLE
[Question]: MQTT vs .HTTP event start_time and end_time difference

### DISCUSSION THREAD

**Discussion by glossyio** [blakeblackshear/frigate#10928] [Ask A Question] [answered]
### What is your question:

I am monitoring MQTT events of `type: "end"` to notify me when I have a "completed" event and process that information for determining how long an object was in the field of view and attach it to other readings, so timestamp of start_time and end_time are critical to this.  
I am also requesting the HTTP API (`/api/events/<id>`) for that event to pull back the thumbnail to display on an external dashboard. This is where I noticed the `start_time` and `end_time` for the MQTT `end` do not match the corresponding HTTP API. In the few sample I checked, it looks like the HTTP API is _exactly_ 5-seconds later (greater) than the MQTT end .after times (see below).

Shouldn't these **MQTT type:end .after** and **HTTP event** values be the same?  Is the HTTP API more accurate than the MQTT values?  Am I totally misunderstanding the use of one or both?  

Sample payload:
```
"event_api": {
    "id": "1712785029.806161-7s7dfw",
    "start_time": 1712785024.806161,
    "end_time": 1712785037.604645,
    ...
}
"after": {
    ...
    "id": "1712785029.806161-7s7dfw",
    "start_time": 1712785029.806161,
    "end_time": 1712785032.604645,
    ...
}
```

**Comment by NickM-27** [comment, marked as answer]
this is because there is a default `post_capture` for events which extends the time of the event. the /events mqtt topic does not include this since it is about the object itself not the recording

**Reply by glossyio** [reply]
OH, sorry, I see that now:  the HTTP API has the additional 5 seconds on **both ends** (I thought it was merely shifted back).

So, the HTTP API `start_time` and `end_time` fields represent the object *with* the `post_capture` times.  The MQTT fields represent the actual object.  MQTT sounds like the one I want for my purposes.  
Thank you.
